src/main/java/core/basesyntax/WriteDataImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class WriteDataImpl implements WriteData {
+
+    private String newPath;

```

Comment:
The `newPath` variable is not initialized or set anywhere in the class. Ensure that it is properly initialized before calling the `writeData` method to avoid a `NullPointerException`.